pkg/gadgets/seccomp/gadget_test.go

```diff
@@ -0,0 +1,186 @@
+				Name: "podname",
+			},
+		},
+	}
+	podName = "podname"
+	expectedNextName = "podname-2"
+	nextName = getSeccompProfileNextName(profileList, podName)
+	if nextName != expectedNextName {
+		t.Fatalf("Invalid computation of next name '%s'. Expecting '%s' from the given profile list",
+			nextName, expectedNextName)
+	}
+
+	// There exist a profile with the podname and another with podname-X.
+	profileList = []seccompprofilev1alpha1.SeccompProfile{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "another-name",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "prefix-podname",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "podname",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "podname-2",
+			},
+		},
+	}
+	podName = "podname"
+	expectedNextName = "podname-3"
+	nextName = getSeccompProfileNextName(profileList, podName)
+	if nextName != expectedNextName {
+		t.Fatalf("Invalid computation of next name '%s'. Expecting '%s' from the given profile list",
+			nextName, expectedNextName)
+	}
+
+	// There exist at least one profile with podname-X.
+	profileList = []seccompprofilev1alpha1.SeccompProfile{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "another-name",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "prefix-podname",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "podname-10",
+			},
+		},
+	}
+	podName = "podname"
+	expectedNextName = "podname-11"
+	nextName = getSeccompProfileNextName(profileList, podName)
+	if nextName != expectedNextName {
+		t.Fatalf("Invalid computation of next name '%s'. Expecting '%s' from the given profile list",
+			nextName, expectedNextName)
+	}
+
+	// Ignoring profiles with sintax podname-X where X is not a number.
+	profileList = []seccompprofilev1alpha1.SeccompProfile{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "another-name",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "podname-xa4b5",
+			},
+		},
+	}
+	podName = "podname"
+	expectedNextName = "podname"
+	nextName = getSeccompProfileNextName(profileList, podName)
+	if nextName != expectedNextName {
+		t.Fatalf("Invalid computation of next name '%s'. Expecting '%s' from the given profile list",
+			nextName, expectedNextName)
+	}
+
+	// Another case where function must ignore the profiles with
+	// syntax podname-X where X is not a number.
+	profileList = []seccompprofilev1alpha1.SeccompProfile{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "another-name",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "podname-xa4b5",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "podname-5",
+			},
+		},
+	}
+	podName = "podname"
+	expectedNextName = "podname-6"
+	nextName = getSeccompProfileNextName(profileList, podName)
+	if nextName != expectedNextName {
+		t.Fatalf("Invalid computation of next name '%s'. Expecting '%s' from the given profile list",
+			nextName, expectedNextName)
+	}
```

Comment:
nit: could you create a test with the following objects to check it's generating the highest one + 1? 

```
profileList = []seccompprofilev1alpha1.SeccompProfile{
		{
			ObjectMeta: metav1.ObjectMeta{
				Name: "podname-2",
			},
		},
		{
			ObjectMeta: metav1.ObjectMeta{
				Name: "podname-5",
			},
		},
	}
```

Reply:
🤦🏽 I added it [here](https://github.com/kinvolk/inspektor-gadget/pull/421#issuecomment-989006475) but I didn't add a unit-test for it. Added now.

Thanks!